packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -116,16 +220,87 @@ class BlacklistingTextInputFormatter extends TextInputFormatter {
     return _selectionAwareTextManipulation(
       newValue,
       (String substring) {
-        return substring.replaceAll(blacklistedPattern, replacementString);
+        return substring.splitMapJoin(
+          filterPattern,
+          onMatch: !allow ? (Match match) => replacementString : null,
+          onNonMatch: allow ? (String nonMatch) => nonMatch.isNotEmpty ? replacementString : '' : null,
+        );
       },
     );
   }
 
-  /// A [BlacklistingTextInputFormatter] that forces input to be a single line.
+  /// A [TextInputFormatter] that forces input to be a single line.
+  static final TextInputFormatter singleLineFormatter = FilteringTextInputFormatter.deny('\n');
+
+  /// A [TextInputFormatter] that takes in digits `[0-9]` only.
+  static final TextInputFormatter digitsOnly = FilteringTextInputFormatter.allow(RegExp(r'[0-9]'));
+}
+
+/// Old name for [FilteringTextInputFormatter.deny].
+@Deprecated(
+  'Use FilteringTextInputFormatter.deny instead. '
+  'This feature was deprecated after v1.20.0-1.0.pre.'
+)
+class BlacklistingTextInputFormatter extends FilteringTextInputFormatter {
+  /// Old name for [FilteringTextInputFormatter.deny].
+  @Deprecated(
+    'Use FilteringTextInputFormatter.deny instead. '
+    'This feature was deprecated after v1.20.0-1.0.pre.'
+  )
+  BlacklistingTextInputFormatter(
+    Pattern blacklistedPattern, {
+    String replacementString = '',
+  }) : super.deny(blacklistedPattern, replacementString: replacementString);
+
+  /// Old name for [filterPattern].
+  @Deprecated(
+    'Use filterPattern instead. '
+    'This feature was deprecated after v1.20.0-1.0.pre.'
+  )
+  Pattern get blacklistedPattern => filterPattern;
+
+  /// Old name for [FilteringTextInputFormatter.singleLineFormatter].
+  @Deprecated(
+    'Use FilteringTextInputFormatter.singleLineFormatter instead. '
+    'This feature was deprecated after v1.20.0-1.0.pre.'
+  )
   static final BlacklistingTextInputFormatter singleLineFormatter
       = BlacklistingTextInputFormatter(RegExp(r'\n'));
 }
 
+/// Old name for [FilteringTextInputFormatter.allow].
+// TODO(ianh): Deprecate these once the samples are migrated.
```

Comment:
Which samples is this referring to?

Reply:
Maybe add a link?

Reply:
gallery in samples. they've said they'll fix it when i land this.

Reply:
https://github.com/flutter/gallery/blob/74d9d92883c6ca10742c44df309f065289cf2f55/lib/demos/material/text_field_demo.dart#L225